app/models/course/experience_points_disbursement.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+# @!attribute [rw] reason
+#   @return [String] The reason for the disbursement.
+#     This reason is shared by each experience points record.
+# @!attribute [r] course_users
+#   @return [Array<CourseUser>] Potential awardees for the given course.
+#     This is used primarily for the rendering of the disbursement form.
+# @!attribute [r] experience_points_records
+#   @return [Array<Course::ExperiencePointsRecords>] Records that will potentially be created.
+#     This reader makes available the records to the controller for authorization.
+# @!attribute [r] recipient_count
+#   @return [Integer] The number of recipients who have been successfully awarded points.
+class Course::ExperiencePointsDisbursement
+  include ActiveModel::Model
+  include ActiveModel::Validations
+
+  attr_accessor :reason
+  attr_reader :course_users, :experience_points_records, :recipient_count
+
+  validates :reason, presence: true
+
+  # Given a course, builds an empty experience points records for each of its
+  # course students.
+  #
+  # @param [Course] course The course the disbursement is for
+  # @return [Course] The course the disbursement is for
+  def course=(course)
+    @course_users = course.course_users.students.with_approved_state.
+                    includes(:experience_points_records)
```

Comment:
I eager loaded because there seems to be a N+1 query on line 31. What should be the proper way to do this?